00XX-common-ledger-protocol/00XX-common-ledger-protocol.md

```diff
@@ -0,0 +1,273 @@
+side protocols carried by this packet.
+
+5. **Side Protocol Data:** A list of key-value pairs, where the key is a
+length-prefixed ascii string, and the value is a length-prefixed octet string.
+The key signifies the side protocol's name, and the value is the data.
+
+## Flow
+
+CLP uses a simple RPC flow. A request-type CLP packet is sent, and a
+response-type CLP packet is sent in response with the same request ID.
+
+If a malformed CLP packet is sent, there should be no response. If a
+response-type CLP packet is sent without matching a previous request, it should
+be ignored.  If a request has already been responded to and another response is
+sent, the second response should be ignored.
+
+Because it would be too slow to atomically save all requestIds that are
+processed, they are not idempotent. It is the responsibility of the requestor
+to make sure they don't duplicate requestIds. The implementation should ensure
+that no two in-flight requests are sent out with the same requestId. The
+responder should always send back a response to a request with the same
+requestId.
+
+The request types are `Message`, `Prepare`, `Fulfill` and `Reject`, and the
+response types are `Ack` (which may be [removed]), `Response`, and `Error`.
+
+### Message
+
+`Message` is used for sending information to the peer. It contains no message
+data, only side protocol data. ILQP packets are attached under the protocol
+`ilp`.
+
+#### Response
+
+- `Response` is returned if the peer acknowledges the `Message`. If the peer
+  has data to send in reply (e.g. a quote response), it is carried in the side
+protocol data.
+
+- `Error` is returned if the peer is not able to process the message, or there
+  was an unexpected error and further messages should not be sent.
+
+### Prepare
+
+`Prepare` is used to create a transfer on the bilateral ledger. The message
+data contains `transferId`, `amount`, `executionCondition`, and `expiresAt`.
+`Prepare` is a request with side effects, because it creates a transfer with
+the given details. This transfer begins in the `prepared` state.
+
+`transferId` is a 128-bit unique ID that references the transfer. `amount` is
+a 64-bit integer denominated in ledger base units. The ledger base units can be
+anything, so long as both parties on the bilateral ledger agree on their
+meaning. Examples of base units are "micro-XRP," "satoshi," or "nano-dollars
+settled over paypal." `executionCondition` is a 256-bit integer containing the
+SHA-256 hash of this conditional transfer's fulfillment.  `expiresAt` is a
+ASN.1 UTC GeneralizedTime containing the expiry of this transfer.
+
+ILP payment packets are attached to the side protocol data under the protocol
+`ilp`. 
+
+#### Response
+
+- `Response` is returned if the peer acknowledges the `Prepare`. This means the
+  transfer is now `prepared` and has been applied to the balance. It may carry
+side protocol data, for example a payment channel claim.
```

Comment:
in this case, 'side' is actually accurate, maybe we should say 'it may carry protocolData, but only for "side protocols", that is, protocols other than the one used in the prepare. Or just keep Ack, that would be simpler ;)